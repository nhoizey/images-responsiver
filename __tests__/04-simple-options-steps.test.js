'use strict';

const prettier = require('prettier');
const imagesResponsiver = require('../index');

function cleanHtml(html) {
	return prettier.format(html, { parser: 'html' });
}

describe('image with simple step options', () => {
	test('simple image', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				minWidth: 120,
				maxWidth: 320,
				steps: 3,
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('simple image with bad number of steps', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				steps: 1,
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('simple image with minWidth = maxWidth', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				minWidth: 320,
				maxWidth: 320,
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('simple image with minWidth > maxWidth', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				minWidth: 640,
				maxWidth: 320,
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});
});
