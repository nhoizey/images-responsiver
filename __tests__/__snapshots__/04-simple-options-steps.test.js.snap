// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image with simple step options simple image 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-120.png 120w, test-220.png 220w, test-320.png 320w\\"
      src=\\"test-640.png\\"
    />
  </body>
</html>
"
`;

exports[`image with simple step options simple image with bad number of steps 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-320.png 320w, test-1280.png 1280w\\"
      src=\\"test-640.png\\"
    />
  </body>
</html>
"
`;

exports[`image with simple step options simple image with minWidth = maxWidth 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-320.png 320w\\"
      src=\\"test-640.png\\"
    />
  </body>
</html>
"
`;

exports[`image with simple step options simple image with minWidth > maxWidth 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"
        test-320.png 320w,
        test-400.png 400w,
        test-480.png 480w,
        test-560.png 560w,
        test-640.png 640w
      \\"
      src=\\"test-640.png\\"
    />
  </body>
</html>
"
`;
